### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by adel-gu**
# For this project I implemented the features listed below:

### Interactions

- Splash screen:
    - Create the Splash screen for welcoming the user, which contains the signup and log-in links.

- Sign up and login pages:
    - The user register in the app with full name, email, and password.
    - The user can log into the app using email and password.
    - If the user is not logged in, they can't access the main pages of the app.

- Home page (categories page)
    - When the user logs in, the app redirects to the categories page.
    - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
    - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
    - There is a button "add a new category" at the bottom that allows the user to create a new category.

- Transactions page
    - For a given category, the list of transactions is presented, ordered by the most recent.
    - At the top of the page the user could see the total amount for the category.
    - There is a button "add a new transaction" at the bottom that allows the user to create a new transaction.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
    - The user fills out a form to create a new category, indicating their name and icon.
    - When the "Save" button is clicked a new category is created, and the app redirects to the home page.
    - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
    - The user fills out a form to create a new transaction with:
        - name 
        - amount
        - categories
    - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
    - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing requirements
    - Create unit and integration tests for all models.

- Technical requirements
    - Used Postgres for the database.
    - Used devise for authentication.
    - Validate all user input to make sure that anyone with bad intentions cannot compromise your app.
    - Deployed the app and it is accessible online.

### CODE REVIEW

File: app/views/categories/_category.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    </div>
+    <div class="card-body">
+      <div class="d-flex justify-content-between">
+        <h5 class="card-title"><%= category.name %></h5>
+        <p class="card-text"><strong>$<%= category.category_total_amount %></strong></p>
+      </div>
+      <p class="position-relative" style="top: 1em"><%= category.updated_at %></p>
+    </div>
+  </div>
+<% end %>
```
Comment (VICTOR-CHUKWU) [inline app/views/categories/_category.html.erb]: - [x] nice job, just a few thing to adjust in your design, category icon is currently overflowing out of the category card, kindly fix that, also make your date to be in human readable format
Screenshot from 2023-02-16 15-00-41

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @adel-gu , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights
- [ ] Test passed
- [ ] project deployed succesfully
- [ ] Good implementation of Guidline

## Suggested changes
_Check the comments under the review._

- [x] kindly add a video of maximum length 5 mins describing your project, you can add the link to the video in your README file or in your PR
- [x]  also give  appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by adel-gu** [comment]
Thank you very much @VICTOR-CHUKWU 👏👏👏, I'll definitely do so 👍

**Review by cynthiainga — APPROVED** [review]
Hi @adel-gu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Good design 👍 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by adel-gu** [comment]
Thank you @cynthiainga 👏👏👏

**Pull request merged**